File: data/multilang_comparison/program_julia/src/func5.jl
include("program_julia_head.jl")
function func5(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array121 = nothing
   if pCounter > 1
       array121 = vars.data[pCounter]
       pCounter -= 1
       array121.refC += 1
   else
       array121 = JArray(zeros(Int, 506), 506, 1, 121)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array121)
   params0.size += 1
   array122 = func10(params0, get_path(), loopsFactor);
   loopLimit36 = (50)/3 + 1
   for i in 1:loopLimit36 
      for i in 0:(array121.size-1) 
         array121.data[i+1] += 1
      end
   end
   for i in 1:array121.size
       if array121.data[i] == 28
               return array121
      end
   end
   array121.refC -= 1
   if array121.refC == 0
       empty!(array121.data)
       array121 = nothing
   end
   return array122;
end

File: data/multilang_comparison/program_cpp_array/src/func5.cpp
#include "program_cpp_array.hpp" 
Array* func5(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array121;
   if (pCounter > 0) {
      array121 = vars->data[--pCounter];
      array121->refC++;
      DEBUG_COPY(array121->id);
   } else {
      array121 = new Array();
      array121->size = 506;
      array121->refC = 1;
      array121->id = 121;
      array121->data = new unsigned int[array121->size]();
      DEBUG_NEW(array121->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array121;
   Array* array122 = func10(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array122->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop36 = 0;
   unsigned int loopLimit36 = (50)/3 + 1;
   for(; loop36 < loopLimit36; loop36++) {
      for (int i = 0; i < array121->size; i++) {
         array121->data[i]++;
      }
   }
   for (int i = 0; i < array121->size; i++) {
       if (array121->data[i] == 28) {
           return array121;
       }
   }
   array121->refC--;
   if(array121->refC == 0) {
      delete[] array121->data;
      array121->data = nullptr;
      DEBUG_FREE(array121->id);
      delete array121;
   }
   return array122;
}

